recipe/meta.yaml

```diff
@@ -73,21 +68,51 @@ test:
     - pytest-xdist         # [CI != "travis"]
     - hypothesis
     - setuptools <60.0.0
-    # some linux tests need a compiler
-    - {{ compiler('c') }}  # [linux]
+    - typing_extensions
+    # some linux tests need a C compiler;
+    # extra f2py tests need a fortran compiler
```

Comment:
This change makes sense. For the record: it was always needed for some of the optional tests. Test failures without a Fortran compiler would be a bug though - and one that I believe we fixed recently after a regression.

Reply:
`cffi` is fast, `mypy` tests are slow. I don't mind much either way; they're not likely to turn up new failures, so the cost-benefit here is a bit of a toss-up.

Reply:
OK, I did a selection based on what I considered (not) relevant, leaving out only:
* mypy: very unlikely that the packaging could break anything about typing
* python-cov: not relevant here as we're not comparing coverage to anything
* wheel: guessing that this has no relevance for conda-forge either

I wrote "mostly optional dependencies" because now `typing_extensions` is a "hard" test requirement (for a single test). Not sure if that's intentional, but I don't mind either way.